bip-0009.mediawiki

```diff
@@ -120,7 +124,7 @@ adding three to the current calendar year when drafting the soft fork proposal):
 
 If the soft fork still not ''locked-in'' and the
 GetMedianTimePast() of a block following a retarget period is at or
-past this timeout, miners should cease setting this bit.
+past this timeout, miners SHOULD cease setting this bit.
```

Comment:
Eric Lombrozo <notifications@github.com> writes:
>> @@ -120,7 +124,7 @@ adding three to the current calendar year when drafting the soft fork proposal):
>>  
>>  If the soft fork still not ''locked-in'' and the
>>  GetMedianTimePast() of a block following a retarget period is at or
>> -past this timeout, miners should cease setting this bit.
>> +past this timeout, miners SHOULD cease setting this bit.
>
> I guess the bigger point here is that we need to draw a distinction between behaviors that will result in invalid blocks vs. behaviors that will still produce valid blocks although we discourage them.

Simple rules win.  An implementer reading this doesn't need to know how
bad things will break if they screw up, just what to do.

> Unfortunately, we don't really have a good way to enforce the correct application of rules by miners, so about the best we can hope for right now is to reduce as much as possible any incentives for miners to feed misleading signals and to detect when miners are in fact feeding misleading signals or not enforcing the rules to allow for intervention.

All language is simply words.  We have no enforcement ability :(

As an aside, I feel RFC 2119 reflected a reduction in flexibility and
increase in dogmatism for IETF.  Shouting at people feels wrong, even if
the aim is clarity, and a turn away from the original peer-like humiity
of "Request For Comments".

Cheers,
Rusty.